lib/spack/spack/cmd/common/arguments.py

```diff
@@ -43,6 +53,44 @@ def add_common_arguments(parser, list_of_arguments):
         parser.add_argument(*x.flags, **x.kwargs)
 
 
+def filter_by_tags(specs_or_packages, tags):
+    """Filter the set of specs passed as argument by tags.
+
+    Args:
+        specs_or_packages: list of items to be filtered
+        tags: tags to be used for filtering
+
+    Returns:
+        list of items in specs that have all the tag in tags
+    """
+
+    # If the list is empty, return early
+    if not specs_or_packages:
+        return
+
+    # Else, try to have a uniform access to packages for both list
+    # of specs and list of packages
+    try:
+        # This statement is needed to check if I am dealing
+        # with a list of specs
+        specs_or_packages[0].package
+        package = lambda x: x.package
+    except AttributeError:
+        package = lambda x: x
```

Comment:
I'd just convert the whole try/catch above to this:

```python
packages = [s.package if isinstance(s, Spec) else s
            for s in specs_or_packages]
```

Then work on the `packages` list below rather than using a lambda all the time.  Also this allows you to filter hybrid lists.